explorer/info.go

```diff
@@ -0,0 +1,82 @@
+package explorer
+
+import (
+	"time"
+
+	"go.sia.tech/core/types"
+)
+
+// BlockFacts returns a bunch of statistics about the consensus set as they
+// were at a specific block.
+type BlockFacts struct {
+	Index             types.ChainIndex
+	TotalWork         types.Work
+	Difficulty        types.Work
+	OakWork           types.Work
+	OakTime           time.Duration
+	GenesisTimestamp  time.Time
+	SiafundPool       types.Currency
+	FoundationAddress types.Address
+
+	// Transaction type counts.
+	SpentSiacoinsCount         uint64
+	SpentSiafundsCount         uint64
+	NewSiacoinsCount           uint64
+	NewSiafundsCount           uint64
+	NewFileContractsCount      uint64
+	RevisedFileContractsCount  uint64
+	ResolvedFileContractsCount uint64
+
+	// Facts about file contracts.
+	ActiveContractCost  types.Currency
+	ActiveContractCount uint64
+	ActiveContractSize  uint64
+	TotalContractCost   types.Currency
+	TotalContractSize   uint64
+	TotalRevisionVolume uint64
+}
+
+// LatestBlockFacts returns facts about the latest black.
+func (e *Explorer) LatestBlockFacts() (BlockFacts, bool) {
+	height, err := e.db.BlockHeight()
+	if err != nil {
+		return BlockFacts{}, false
+	}
+	return e.BlockFacts(height)
+}
+
+// BlockFacts returns facts about the black at the the specified height.
+func (e *Explorer) BlockFacts(height uint64) (BlockFacts, bool) {
```

Comment:
should take a `types.ChainIndex`, since it should be possible to request the facts for any block (not just blocks on the best chain)